File: r/nt_solutions/segmentation_3_snakes_levelset/exo8.R




gD <- grad(phi, order=2)
d <- pmax(eps*array(1, c(n,n)), sqrt(apply(gD**2, c(1,2), sum)))
g <- gD/array(rep(d,2), c(dim(d),2))
G <- d*div(g[,,1], g[,,2], order=2) - lambd*(f0-c1)**2 + lambd*(f0-c2)**2

File: python/nt_solutions/segmentation_3_snakes_levelset/exo8.py
gD = grad(phi, order=2)
d = np.maximum(eps*np.ones([n,n]), np.sqrt(np.sum(gD**2, 2)))
g = gD/np.repeat(d[:,:,np.newaxis], 2, 2)
G = d*div(g[:,:,0], g[:,:,1], order=2) - lambd*(f0-c1)**2 + lambd*(f0-c2)**2